src/librustc_yk_sections/mir_cfg.rs

```diff
@@ -112,26 +121,59 @@ impl<'a, 'tcx, 'gcx> ConvCx<'a, 'tcx, 'gcx> {
     }
 }
 
-/// Converts and serialises the specified DefIds, returning an linkable ELF object.
+/// Writes TIR to file for the specified DefIds, possibly returning a linkable ELF object.
 pub fn generate_tir<'a, 'tcx, 'gcx>(
-    tcx: &'a TyCtxt<'a, 'tcx, 'gcx>, def_ids: &DefIdSet, exe_filename: PathBuf, dump_filename: Option<PathBuf>)
-    -> Result<YkExtraLinkObject, Box<dyn Error>> {
+    tcx: &'a TyCtxt<'a, 'tcx, 'gcx>, def_ids: &DefIdSet, mode: TirMode)
+    -> Result<Option<YkExtraLinkObject>, Box<dyn Error>>
+{
+    let tir_path = do_generate_tir(tcx, def_ids, &mode)?;
+    match mode {
+        TirMode::Default(_) => {
+            // In this case the file at `tir_path` is a raw binary file which we use to make an
+            // object file for linkage.
+            let obj = YkExtraLinkObject::new(&tir_path, SECTION_NAME);
+            fs::remove_file(tir_path)?; // Now we have our object, we can remove the temp file.
```

Comment:
We might want to suppress error if the temporary file can't be removed (it's not the end of the world)?

Reply:
Up to you. It's super unlikely to happen.